### ISSUE TITLE
FileSystemDirectoryHandle.resolve is a very vague method name, consider renaming it

### ISSUE THREAD

**Issue by smaug----** [whatwg/fs#4]
"resolve" doesn't hint too much what it is actually doing. One needs to read the algorithm to see what it actually returns.

**Comment by a-sully** [comment]
I agree that the name is not great. Current usage is [still low](https://chromestatus.com/metrics/feature/timeline/popularity/3178) so I'm not necessarily opposed to a name change, but it would be a bit of a hassle to deprecate the current name. Do you think the benefits of renaming this method would be worth that hassle?

If so, the next question is what this should be named instead. Something like `getRelativePath()` or `getRelativePathComponents()` is more descriptive of what's happening here (since we don't actually return the path, but an array of path components). While we're at it, reversing the parameter order might make more sense...

Current: `dir.resolve(file)`

Alternatives:
- `file.getRelativePath(dir)`
- `file.getRelativePathComponents(dir)`
- `file.getPathRelativeTo(dir)`
- others?

**Comment by poirierlouis** [comment]
Hello, I don't know if I'm trespassing by commenting here. Feel free to remove this if necessary.

> Do you think the benefits of renaming this method would be worth that hassle?

IMHO, it would be worth, as `resolve` is not particularly explicit. Someone discovering the API would need to read the documentation to be certain of what this method does. I believe alternatives proposed are more explicit when discovering this API, as 'path' and 'relative' makes direct sense in API's context.

I also like `file.getPathRelativeTo(dir)`.

> While we're at it, reversing the parameter order might make more sense...

Do you mean it would be called like this `dir.getPathRelativeTo(file)` / `dir.resolve(file)`?

**Comment by annevk** [comment]
Unfortunately it seems the ball was dropped on this one. I don't see us renaming this at this point.
